Classes/Controllers/PBGitCommitController.m

```diff
@@ -239,6 +240,58 @@ - (IBAction) refresh:(id) sender
 	[repository reloadRefs];
 }
 
+- (IBAction) prepareCommitMessage:(id) sender
+{
+	NSError *error;
+	NSString *messagePath = [NSTemporaryDirectory() stringByAppendingPathComponent:[@"commit-" stringByAppendingString:[[NSUUID UUID] UUIDString]]];
+	NSMutableArray<NSString *> *arguments = [NSMutableArray arrayWithObject:messagePath];
+
+	if ([[[self repository] index] isAmend]) {
+		// git itself seems to pass HEAD when doing a plain 'git commit --amend', but git's documentation says the third argument can be a commit hash
+		[arguments addObject:@"commit"];
+		[arguments addObject:self.repository.headOID.SHA];
+
+		// Write the message of the commit we're amending to a file so we can run the prepare-commit-msg hook on it
+		GTReference *headRef = [self.repository.gtRepo headReferenceWithError:NULL];
+		GTCommit *commit = [headRef resolvedTarget];
+
+		[commit.message writeToFile:messagePath atomically:YES encoding:NSUTF8StringEncoding error:NULL];
+	}
+
+	self.isBusy = YES;
+
+	if ([repository executeHook:@"prepare-commit-msg" arguments:arguments error:&error]) {
```

Comment:
This goes slightly against my current goal of keeping the git-wrangling contained inside non-controller classes (so it's easier to swap out with libgit2). Here, it's not that important, but I feel like the "guts" of this method should be moved to (maybe, because of the notification reuse below ?) `PBGitIndex`, so this one is only concerned with "being busy" and "preparing commit message".